src/pages/[SbtAddress].tsx

```diff
@@ -0,0 +1,282 @@
+import Connect from "@/components/connection/connect";
+import styles from "@/styles/generator.module.css";
+import sbt_abi from "@/abi/starknet/sbt_abi.json";
+import { TextField } from "@mui/material";
+import LoadingScreen from "@/components/UI/screens/loadingScreen";
+
+type MetadataProps = {
+  name: string;
+  description: string;
+  image: string;
+};
+
+type CallDataProps = {
+  calldata: (string | number)[];
+  contractAddress: string;
+  entrypoint: string;
+};
+
+type TransactionDatas = {
+  status: string;
+};
+
+const SbtGenerator: NextPage = () => {
+  const router = useRouter();
+  const { SbtAddress } = router.query;
+
+  const { account, address } = useAccount();
+  const [contractAddress, setContractAddress] = useState<string>("");
+  const [privateKey, setPrivateKey] = useState<BN>();
+  const [tokenId, setTokenId] = useState<string>("0");
+  const [password, setPassword] = useState<string>("");
+  const [sbtData, setSbtData] = useState<MetadataProps>();
+  const [callData, setCallData] = useState<CallDataProps[]>();
+  const [element, setElement] = useState<ReactElement | null>(null);
+  const [loadingMessage, setLoadingMessage] = useState<string>("");
+  const [clickedMint, setClickedMint] = useState<boolean>(false);
+  const [transactionHash, setTransactionHash] = useState<string>("");
+  const [finalStep, setFinalStep] = useState<boolean>(false);
+  const {
+    data: transactionData,
+    loading: transactionLoading,
+    error: transactionError,
+  } = useTransaction({ hash: transactionHash });
+
+  const { execute } = useStarknetExecute({
+    calls: callData,
+  });
+  const { contract } = useContract({
+    address: contractAddress,
+    abi: sbt_abi,
+  });
+  const { data, loading, error } = useStarknetCall({
+    contract,
+    method: "get_uri",
+    args: [1],
+    options: {
+      watch: false,
+    },
+  });
+
+  useEffect(() => {
+    setContractAddress(SbtAddress as string);
+  }, [SbtAddress]);
+
+  useEffect(() => {
+    if (!loadingMessage) setElement(null);
+    setElement(<LoadingScreen message={loadingMessage} />);
+  }, [loadingMessage]);
+
+  useEffect(() => {
+    if (!loading) setElement(null);
+    if (address && loading && !error) setLoadingMessage("Fetching SBT data");
+  }, [loading, address]);
+
+  useEffect(() => {
+    if (!transactionData || !clickedMint) return;
+    const status = (
+      transactionData as GetTransactionResponse & TransactionDatas
+    ).status;
+    setLoadingMessage(status);
+    if (status === "ACCEPTED_ON_L2" || status === "ACCEPTED_ON_L1") {
+      setLoadingMessage("");
+      setFinalStep(true);
+    }
+  }, [transactionData, transactionLoading, transactionError]);
+
+  useEffect(() => {
+    if (!sbtData && data) {
+      let dataUri = "";
+      data[0].forEach((c: number) => {
+        dataUri += String.fromCharCode(c);
+      });
+      const metadataUri = dataUri.replace("ipfs://", "https://ipfs.io/ipfs/");
```

Comment:
is this the best way to access the data ? 

Cause using `ipfs.io/ipfs` is very slow in general. If you don't know any other it's not worth searching atm :).

Reply:
Usually I use Pinata to fetch but recently I got errors when fetching metadata through it. Can retry Pinata and see how it goes